.travis.yml

```diff
@@ -1,40 +1,98 @@
 sudo: required
+_aliases:
+  - &ui-tests
+    addons:
+      firefox: latest-nightly
+      hosts: example.com
+    env: TOXENV=discopane-ui-tests DISPLAY=:99.0 GECKODRIVER=0.19.1
+    before_install:
+      - sudo apt-get -qq update
+      - sudo apt-get install jwm
+      - wget -O /tmp/geckodriver.tar.gz https://github.com/mozilla/geckodriver/releases/download/v$GECKODRIVER/geckodriver-v$GECKODRIVER-linux64.tar.gz
+      - mkdir $HOME/geckodriver && tar xvf /tmp/geckodriver.tar.gz -C $HOME/geckodriver
+      - export PATH=$HOME/geckodriver:$PATH
+      - firefox --version
+      - geckodriver --version
+      - pip install tox
+    before_script:
+      - yarn install
+      - sh -e /etc/init.d/xvfb start
+      - sleep 5
```

Comment:
the old code did `sleep 3` -- could we try going back to that or were you running into issues? The tests need to be as fast as possible.